datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxHorizontalScalingService.java

```diff
@@ -99,46 +123,86 @@ public String triggerScalingFlow(SdxCluster sdxCluster, StackScaleV4Request scal
     }
 
     public void validateHorizontalScaleRequest(SdxCluster sdxCluster, DatalakeHorizontalScaleRequest request) {
-        if (!ENTERPRISE.equals(sdxCluster.getClusterShape())) {
-            throw new BadRequestException(String.format("scaling not supported on: %s. " +
-                    "Only the ENTERPRISE Datalake shape support the horizontal scaling functionality.", sdxCluster.getClusterShape().name()));
+        if (ENTERPRISE != sdxCluster.getClusterShape()) {
+            throw new BadRequestException(String.format("Horizontal scaling not supported on: %s. Please use ENTERPRISE Data lake shape",
+                    sdxCluster.getClusterShape().name()));
         }
         String hostGroup = request.getGroup();
         if (!DatalakeInstanceGroupScalingDetails.valueOf(hostGroup.toUpperCase(Locale.ROOT)).isScalable()) {
-            throw new BadRequestException(String.format("%s host group cannot scale", hostGroup));
+            throw new BadRequestException(String.format("%s hostgroup cannot scale", hostGroup));
         }
         DetailedEnvironmentResponse environmentResponse = environmentClientService.getByName(sdxCluster.getEnvName());
         if (!EnvironmentStatus.upscalable().contains(environmentResponse.getEnvironmentStatus())) {
-            throw new BadRequestException(String.format("The DAtalake cannot be scaled because the environment %s is in the %s status.",
-                    sdxCluster.getEnvName(), environmentResponse.getEnvironmentStatus().name()));
+            throw new BadRequestException(String.format("The Data lake cannot be scaled in the %s environment status.",
+                    environmentResponse.getEnvironmentStatus().name()));
         }
         StackV4Response stackResponse = stackV4Endpoint.getByCrn(1L, sdxCluster.getStackCrn(), Set.of());
-        String nodeRequestValidity = isNodeRequestValid(stackResponse, request);
-        if (StringUtils.isNotEmpty(nodeRequestValidity)) {
-            throw new BadRequestException(String.format("The requested nodeCount does not reach the minimum condition of horizontal scaling. Issues: %s",
-                    nodeRequestValidity));
+        if (YARN == stackResponse.getCloudPlatform()) {
+            throw new BadRequestException("The Data lake horizontal scaling is not supported on YCloud");
+        }
+        String nodeRequestValid = isNodeRequestValid(stackResponse, request);
+        if (StringUtils.isNotEmpty(nodeRequestValid)) {
+            throw new BadRequestException(nodeRequestValid);
         }
     }
 
     private String isNodeRequestValid(StackV4Response stack, DatalakeHorizontalScaleRequest request) {
```

Comment:
I would refactor this method:
`private Optional<String> validateNodeRequest`

since currently the name indicates the return type is boolean but it is string, also you could check the optional instead of the string itself